CefSharp.Core/Internals/CefBrowserHostWrapper.cpp

```diff
@@ -301,6 +301,27 @@ void CefBrowserHostWrapper::SendKeyEvent(int message, int wParam, int lParam)
     }
     keyEvent.modifiers = GetCefKeyboardModifiers(wParam, lParam);
 
+    // mimic alt-gr check behaviour from
+    // src/ui/events/win/events_win_utils.cc: GetModifiersFromKeyState
+    if ((keyEvent.type == KEYEVENT_CHAR) && IsKeyDown(VK_RMENU))
```

Comment:
We should be able to avoid the additional `KEYEVENT_CHAR` check if we move this code into the else branch above. 

```c#
keyEvent.modifiers = GetCefKeyboardModifiers(wParam, lParam);

if (message == WM_KEYDOWN || message == WM_SYSKEYDOWN)
{
	keyEvent.type = KEYEVENT_RAWKEYDOWN;
}
else if (message == WM_KEYUP || message == WM_SYSKEYUP)
{
	keyEvent.type = KEYEVENT_KEYUP;
}
else
{
	keyEvent.type = KEYEVENT_CHAR;

	// mimic alt-gr check behaviour from
	// src/ui/events/win/events_win_utils.cc: GetModifiersFromKeyState
	if (IsKeyDown(VK_RMENU))
	{
		// reverse AltGr detection taken from PlatformKeyMap::UsesAltGraph
		// instead of checking all combination for ctrl-alt, just check current char
		HKL current_layout = ::GetKeyboardLayout(0);

		// https://docs.microsoft.com/en-gb/windows/win32/api/winuser/nf-winuser-vkkeyscanexw
		// ... high-order byte contains the shift state,
		// which can be a combination of the following flag bits.
		// 2 Either CTRL key is pressed.
		// 4 Either ALT key is pressed.
		SHORT scan_res = ::VkKeyScanExW(wParam, current_layout);
		if (((scan_res >> 8) & 0xFF) == (2 | 4)) // ctrl-alt pressed
		{
			keyEvent.modifiers &= ~(EVENTFLAG_CONTROL_DOWN | EVENTFLAG_ALT_DOWN);
			keyEvent.modifiers |= EVENTFLAG_ALTGR_DOWN;
		}
	}
}
```

I haven't tested this code though.

Reply:
Adjusted the code accordingly and tested.

Reply:
Thanks